public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.test.ts

```diff
@@ -797,4 +802,126 @@ describe('IN_LOGFMT completions', () => {
       ]
     `);
   });
+
+  describe('calculateRange', () => {
+    let monaco: Monaco;
+    beforeEach(() => {
+      monaco = {
+        Range: {
+          lift(range: monacoTypes.Range): monacoTypes.Range {
+            return range;
+          },
+        },
+      } as Monaco;
+    });
+
+    it('getSituation fails to return autocomplete when inserting before any other labels', () => {
+      // Note this is not desirable functionality
+      const situation: Situation | null = getSituation('{label1="value1",label2="value2"}', 1);
+      expect(situation).toBe(null);
+    });
+    it('tests inserting new label before existing label name', () => {
+      const situation: Situation | null = getSituation('{label1="value1",label2="value2"}', 17);
+      expect(situation?.type).toBe('IN_LABEL_SELECTOR_NO_LABEL_NAME');
+      const word: monacoTypes.editor.IWordAtPosition = {
+        word: 'label2="value2"',
+        startColumn: 17,
+        endColumn: 32,
+      };
+      const wordUntil: monacoTypes.editor.IWordAtPosition = {
+        word: '',
+        startColumn: 17,
+        endColumn: 17,
+      };
+      const position: monacoTypes.Position = {
+        lineNumber: 1,
+        column: 17,
+      } as monacoTypes.Position;
+
+      expect(calculateRange(situation, word, wordUntil, monaco, position)).toMatchObject({
+        startLineNumber: 1,
+        endLineNumber: 1,
+        startColumn: 17,
+        endColumn: 17,
+      });
+    });
+    it('tests inserting new label within existing label value', () => {
```

Comment:
It seems that there this test is twice here and in line 874.

Reply:
Should be fixed now